wgengine/magicsock/magicsock.go

```diff
@@ -4411,14 +4411,14 @@ func (de *endpoint) addrForWireGuardSendLocked(now mono.Time) (udpAddr netip.Add
 
 	candidates := make([]netip.AddrPort, 0, len(de.endpointState))
 	for ipp := range de.endpointState {
-		if ipp.Addr().Is4() && de.c.noV4.Load() {
-			continue
-		}
-		if ipp.Addr().Is6() && de.c.noV6.Load() {
-			continue
-		}
 		candidates = append(candidates, ipp)
 	}
```

Comment:
if you want, at this point these 4 lines can use https://pkg.go.dev/golang.org/x/exp/maps#Keys and be:

```go
     candidates := maps.Keys(de.endpointState)
```